---
 Makefile.shared |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: openssl-1.1.0g/Makefile.shared
===================================================================
--- openssl-1.1.0g.orig/Makefile.shared
+++ openssl-1.1.0g/Makefile.shared
@@ -154,7 +154,7 @@ LINK_SO_SHLIB_UNPACKED=	\
 DETECT_GNU_LD=($(CC) -Wl,-V /dev/null 2>&1 | grep '^GNU ld' )>/dev/null
 
 DO_GNU_SO_COMMON=\
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$(SHLIBNAME_FULL)"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-soname=$(SHLIBNAME_FULL)"
 DO_GNU_DSO=\
 	$(DO_GNU_SO_COMMON)
 DO_GNU_SO=\
